README.md

```diff
@@ -1,3 +1,255 @@
 [![CI](https://github.com/couler-proj/couler/workflows/CI/badge.svg)](https://github.com/couler-proj/couler/actions?query=event%3Apush+branch%3Amaster)
 
-# couler
+# Couler
+
+## What is Couler?
+
+Couler aims to provide a unified interface for constructing and managing workflows on
+different workflow engines, such as [Argo Workflows](https://github.com/argoproj/argo) and [Apache Airflow](https://airflow.apache.org/).
+
+## Why use Couler?
+
+Many workflow engines exist nowadays, e.g. [Apache Airflow](https://airflow.apache.org/), [Kubeflow Pipelines](https://github.com/kubeflow/pipelines), and [Argo Workflows](https://github.com/argoproj/argo).
```

Comment:
Not sure whether we should treat kfp as an engine here, as it is a layer atop engines. One other example can be included here is Tekton.